x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/AliasTests.java

```diff
@@ -0,0 +1,86 @@
+import org.elasticsearch.xpack.esql.core.type.EsField;
+import org.elasticsearch.xpack.esql.expression.function.ReferenceAttributeTests;
+import org.elasticsearch.xpack.esql.expression.function.UnsupportedAttribute;
+import org.elasticsearch.xpack.esql.io.stream.PlanNameRegistry;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.in;
+
+public class AliasTests extends AbstractWireTestCase<Alias> {
+    @Override
+    protected Alias createTestInstance() {
+        Source source = SourceTests.randomSource();
+        String name = randomAlphaOfLength(5);
+        String qualifier = randomBoolean() ? null : randomAlphaOfLength(3);
+        // TODO better randomChild
+        Expression child = ReferenceAttributeTests.randomReferenceAttribute();
+        boolean synthetic = randomBoolean();
+        return new Alias(source, name, qualifier, child, new NameId(), synthetic);
+    }
+
+    @Override
+    protected Alias mutateInstance(Alias instance) throws IOException {
+        Source source = instance.source();
+        String name = instance.name();
+        String qualifier = instance.qualifier();
+        Expression child = instance.child();
+        boolean synthetic = instance.synthetic();
+        switch (between(0, 3)) {
+            case 0 -> name = randomAlphaOfLength(name.length() + 1);
+            case 1 -> qualifier = randomValueOtherThan(qualifier, () -> randomBoolean() ? null : randomAlphaOfLength(3));
+            case 2 -> child = randomValueOtherThan(child, ReferenceAttributeTests::randomReferenceAttribute);
+            case 3 -> synthetic = false == synthetic;
+        }
+        return new Alias(source, name, qualifier, child, instance.id(), synthetic);
+    }
+
+    @Override
+    protected Alias copyInstance(Alias instance, TransportVersion version) throws IOException {
+        return copyInstance(
+            instance,
+            getNamedWriteableRegistry(),
+            (out, v) -> new PlanStreamOutput(out, new PlanNameRegistry(), null).writeNamedWriteable(v),
+            in -> {
+                PlanStreamInput pin = new PlanStreamInput(in, new PlanNameRegistry(), in.namedWriteableRegistry(), null);
+                Alias deser = (Alias) pin.readNamedWriteable(NamedExpression.class);
+                assertThat(deser.id(), equalTo(pin.mapNameId(Long.parseLong(instance.id().toString()))));
+                return deser;
+            },
+            version
+        );
+    }
+
+    @Override
+    protected final NamedWriteableRegistry getNamedWriteableRegistry() {
+        List<NamedWriteableRegistry.Entry> entries = new ArrayList<>(NamedExpressions.getNamedWriteables());
+        entries.addAll(Attribute.getNamedWriteables());
+        entries.add(UnsupportedAttribute.ENTRY);
```

Comment:
Noting that `UnsupportedAttribute` attribute is present twice in this registry, once with an entry that reads an `Attribute` (as initialised in own `ENTRY`) and once that reads a `NamedExpression` (as created in `NamedExpressions#getNamedWriteables()`). I guess it's intentional and wondering why.

Reply:
It has two category classes. Which *is* funky, but it's kind of how we serialize in ESQL - we sometimes read and write `Attribute` and sometimes `NamedExpression` and sometimes `Expression`. They all could be an `UnsupportedAttribute`. So I'm making a `categoryClass` for each of those three. Well, those one makes it for the first two, because we don't yet support `NamedWriteable` for `Expression` - that's next I think. But there are lots of `Expression` subclasses so that'll take a long while.